### ISSUE TITLE
Текущие корректировки и проблемы

### ISSUE THREAD

**Issue by EgorGologuzov** [EgorGologuzov/obyavi.site#36]
Здесь будем обсуждать корректировки, которые нужно внести в проект. Это место должно быть намного лучше комментариев в гугл таблицах.

**По оперативным вопросам сюда не пишем, пишем в ТГ или ВК.**

**Comment by EgorGologuzov** [comment]
# Список проблем

## 1.  [Fixed] Дублирование объявлений в AdSearchClientPage

(тут же куча ошибок в консоли из-за дублирования ключей)

**Причины:** Скорее всего возникает, из-за того, что ты вызываешь adService.getMyAds с одним и тем же аргументом page.
**Решение:** Отслеживать какие страницы уже загружены и не загружать их повторно.

## 2.  Странное поведение при нажатии на кнопку поиска в AdSearchClientPage

Если так и задумано, что объявления выстраиваются в одну линию, то надо убрать надпись "Нет элементов". Вообще лучше сделать, чтобы найденные объявления загружались списком как и предложенные.

## 3.  Дизайн в AdSearchClientPage

1. Убери надпись найдено n из n, и выведи просто сколько объявлений найдено. Сделай поменьше шрифт.
3. Сделай адаптив кнопок фильтра и сортировки под маленький и средний экран.

## 4.  [Fixed] Аватар в EditProfilePageClient

Сделай возможность выбрать картинку из фалов на компе и подставить ее в аватар. Как это сделать можешь посмотреть у меня в CreateAdPageClient и EditAdPageClient.

## 5.  [Fixed] Кнопки в ProfilePageClient

В мобильной версии кнопки "Мои отзывы" и "Избранные объявления" тоже должны быть видны

## 6.  [Fixed] Отображение данных профилей в ListOfClientReviewsClient и ClientPageClient

Используй мои компонент ProfileInfo для отображения данных профилей. Посмотри как его использовать на странице AdPageClient. Скопируй и измени этот компонент, чтобы добавить Онлайн метку, дату или изменить размер.

#  Не торопись
@Flaric123 
Не нужно гнаться за количеством страниц. Удели внимание уже начатым тобой страницам следующие две недели. Можно делать по 1-2 страницы за этот срок, главное хорошо прорабатывать их.

**Comment by EgorGologuzov** [comment]
# P.S.
Пиши сюда какие проблемы ты исправил, когда исправишь

**Comment by Flaric123** [comment]
1. Двойной рендер из-за Strict мода в девелопменте. В продакшене двойного рендера не будет. Можешь проверить - замени React.StrictMode на React.Fragment в index.js

**Comment by EgorGologuzov** [comment]
> 1. Двойной рендер из-за Strict мода в девелопменте. В продакшене двойного рендера не будет. Можешь проверить - замени React.StrictMode на React.Fragment в index.js

Да, все так. Но StrictMode как раз и предназначен для выявления подобных проблем, так как при определенных обстоятельствах неоднократный рендеринг может произойти и в продакшене.

**Решение:**
```jsx
   const hasLoaded = useRef(false);

   useEffect(() => {
       if (!hasLoaded.current) {
           load({ page: 1 });
           hasLoaded.current = true;
       }
   }, []);
```
